spark_client/services.py

```diff
@@ -0,0 +1,845 @@
+        """
+
+        if (service_account.primary is True) and any(
+            [account.primary for account in self.all()]
+        ):
+            self.logger.info(
+                "Primary service account provided. Switching primary account from account"
+            )
+            for account_id, account in self.cache.items():
+                if account.primary is True:
+                    self.logger.debug(
+                        f"Setting primary of account {account.id} to False"
+                    )
+                    account.primary = False
+
+        self.cache[service_account.id] = service_account
+        return service_account.id
+
+    def delete(self, account_id: str) -> str:
+        """Delete the service account associated with the provided id.
+
+        Args:
+            account_id: service account id to be deleted
+        """
+        return self.cache.pop(account_id).id
+
+    def set_primary(self, account_id: str) -> str:
+        """Set the primary account to the one related to the provided account id.
+
+        Args:
+            account_id: account id to be elected as new primary account
+        """
+        if account_id not in self.cache.keys():
+            raise NoAccountFound(account_id)
+
+        if any([account.primary for account in self.all()]):
+            self.logger.info("Switching primary account")
+            for account_id, account in self.cache.items():
+                if account.primary is True:
+                    self.logger.debug(
+                        f"Setting primary of account {account.id} to False"
+                    )
+                    account.primary = False
+
+        self.cache[account_id].primary = True
+        return account_id
+
+    def set_configurations(self, account_id: str, configurations: PropertyFile) -> str:
+        """Set a new service account configuration for the provided service account id.
+
+        Args:
+            account_id: account id for which configuration ought to be set
+            configurations: PropertyFile representing the new configuration to be stored
+        """
+
+        if account_id not in self.cache.keys():
+            raise NoAccountFound(account_id)
+
+        self.cache[account_id].extra_confs = configurations
+        return account_id
+
+
+def parse_conf_overrides(
+    conf_args: List, environ: Dict = dict(os.environ)
+) -> PropertyFile:
+    """Parse --conf overrides passed to spark-submit
+
+    Args:
+        conf_args: list of all --conf 'k1=v1' type args passed to spark-submit.
+            Note v1 expression itself could be containing '='
+        environ: dictionary with environment variables as key-value pairs
+    """
+    conf_overrides = dict()
+    if conf_args:
+        for c in conf_args:
+            try:
+                kv = c.split("=")
+                k = kv[0]
+                v = "=".join(kv[1:])
+                conf_overrides[k] = environ.get(v, v)
```

Comment:
should we use os.path.expandvars instead?

Reply:
I will address this here!